### PR TITLE
Add missing dev depends

### PR DESCRIPTION
**Pull request by smorimoto**

### CODE REVIEW

File: lsp.opam [hunk 1]
```diff
@@ -26,7 +26,8 @@ depends: [
   "dune" {>= "3.0"}
   "jsonrpc" {= version}
   "yojson"
-  "ppx_yojson_conv_lib" {>= "v0.14"}
+  "ppx_yojson_conv" {>= "v0.14"}
```
Comment (rgrinberg) [inline lsp.opam:29]: Do we need the preprocessor? We're only using the runtime

Comment (smorimoto) [reply lsp.opam]: @rgrinberg Oops, I guess we actually needed different changes. Updated PR.

File: lsp/src/dune [hunk 2]
```diff
@@ -4,8 +4,6 @@
  (name lsp)
  (public_name lsp)
  (libraries jsonrpc ppx_yojson_conv_lib uutf yojson)
- (lint
-  (pps ppx_yojson_conv))
```
Comment (rgrinberg) [inline lsp/src/dune:8]: We need this in development to promote the generated ppx derivers

Comment (smorimoto) [reply lsp/src/dune:8]: Should we list that dependency for development?

Comment (rgrinberg) [reply lsp/src/dune:8]: I think that would be correct

Comment (smorimoto) [reply lsp/src/dune]: Fixed.

**Pull request merged**